src/unique_impl.rs

```diff
@@ -83,20 +80,12 @@ impl<I> Iterator for Unique<I>
     type Item = I::Item;
 
     fn next(&mut self) -> Option<I::Item> {
-        while let Some(v) = self.iter.iter.next() {
-            if let Entry::Vacant(entry) = self.iter.used.entry(v) {
-                let elt = entry.key().clone();
-                entry.insert(());
-                return Some(elt);
-            }
-        }
-        None
+        return self.iter.next();
```

Comment:
(Cloning before we check if the element is unique, is another negative performance impact for types where Clone has a measurable cost).